### PR TITLE
DOC: Improve examples of df.append to better show the ignore_index param

### LINKED ISSUE #41407: DOC: Unclear first example in pandas.DataFrame.append
#### Location of the documentation

pandas.DataFrame.append 

#### Documentation problem

I think the example chosen to show the difference between ignore_index=False and ignore_index = True (the very first on the page) could make this difference more clear.

#### Suggested fix for documentation

Choosing a custom index for the first df (the one whose append function is called) could help to especially point out the fact that ignore_index=True actually overwrites the whole index and does not keep the index of the first dataframe, as it could be interpreted from this example (given somebody dared to not read the doc above 😉 )

### PR DESCRIPTION
**Pull request by t0asty**
Change index to [0, 2] to stress that ignore_index=True resets the index of both dataframes

- [x] closes #41407 
- [x] tests passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Test Output: 

```
################################################################################
################################## Validation ##################################
################################################################################

1 Errors found:
	Return value has no description
```

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8811,18 +8811,18 @@ def append(
...
         --------
-        >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'))
+        >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'), index=[0, 2])
         >>> df
            A  B
         0  1  2
-        1  3  4
-        >>> df2 = pd.DataFrame([[5, 6], [7, 8]], columns=list('AB'))
+        2  3  4
+        >>> df2 = pd.DataFrame([[5, 6], [7, 8]], columns=list('AB'), index=[0, 2])
```
Comment (phofl) [inline pandas/core/frame.py:8819]: Could you use non matching indexes?

**Review by phofl — CHANGES_REQUESTED** [review]
Small comment, otherwise lgtm

File: pandas/core/frame.py (hunk 1 above)
Comment (phofl) [reply pandas/core/frame.py]: Sorry, this was not clear enough. Could we do something like ``[0, 4]`` to show that indexes get duplicated when they have the same element and the second element is non-matching to show the other case? Should have made the more clear before, sorry.

Comment (t0asty) [reply pandas/core/frame.py]: Sure, there you go :)

**Review by phofl — APPROVED** [review]
Thanks very much

**Comment by t0asty** [comment]
Anything to fix? Sounds not like it's coming from my changes, does it? 

`Sphinx parallel build error:
nbsphinx.NotebookError: UndefinedError in user_guide/style.ipynb:
'nbformat.notebooknode.NotebookNode object' has no attribute 'tags'`

**Comment by phofl** [comment]
No, see #41450

**Comment by phofl** [comment]
Sorry pressed the wrong button, this was not on purpose

**Review by datapythonista — COMMENTED** [review]
Being honest, I find the original example clearer and more realistic. I think in most cases users will experience appending or concatenating dataframes that both have the default index and start with `0`. And `ignore_index` will be used to get a new default index for both together. That's what the original example illustrates, I find this new one a bit arbitrary.

Not against merging this if other people find it more clear. But if you think the example it's not clear, maybe using meaningful data would make more sense than this change. Like finding two dataframes that the users can relate with and make sense appending. We have some examples with animals, and their number of legs, their maximum speed... And some others that are not just random data.

**Comment by datapythonista** [comment]
Btw, if you merge maser, the CI should be fixed.

**Comment by datapythonista** [comment]
Another btw, feel free to add the `Return` description, since the validation is complaining that one is missing, and should be very easy to add.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -8811,18 +8812,18 @@ def append(
 
         Examples
         --------
-        >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'))
+        >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'), index=[0, 2])
```
Comment (attack68) [inline pandas/core/frame.py:8815]: The comment from @datapythonista finding the original clearer, I agree with, why do you not change the index here to ["x", "y"] and for df2. Then you achieve; highlighting the non-unique element created; avoid any confusion with the same 0,1 and indexes which are only conincidentally the same in the original example; and demonstrate the index has a dtype change rom string to base RangeIndex.

**Review by attack68 — CHANGES_REQUESTED** [review]
very minor comment

File: pandas/core/frame.py (hunk 2 above)
Comment (t0asty) [reply pandas/core/frame.py:8815]: Yes, that sounds good!

**Review by datapythonista — APPROVED** [review]
Thanks @t0asty, lgtm

**Comment by jreback** [comment]
thanks @t0asty

**Pull request merged**